feat #15 : 가게 메뉴 등록 구현

log1995 opened the pull request.

log1995 committed:
가게 메뉴 등록 구현

f-lab-joseph commented on src/main/java/com/yummy/delivery/controller/MenuController.java:
```diff
@@ -0,0 +1,21 @@
+
+    /* 메뉴 등록 */
+    @PostMapping("menu/register/{store_id}")
+    public void menuSet(@PathVariable("store_id") int store_id, Menu menu){
```
camel case 를 지켜주세요

f-lab-joseph commented on src/main/java/com/yummy/delivery/domain/Menu.java:
```diff
@@ -0,0 +1,23 @@
+@ToString
+public class Menu {
+
+    private int id;
```
DTO, ENTITY 에서는 primitive type 대신 wrapper type 을 사용해주세요. 왜 그럴까요 ?

log1995 replied:
primitive type에는 null 값을 넣을 수 없기 때문에 DB 연동 시 해당 컬럼 값에 null이 필요할 경우 사용할 수가 없습니다.

f-lab-joseph commented on src/main/java/com/yummy/delivery/service/MenuService.java:
```diff
@@ -0,0 +1,31 @@
+@Service
+public class MenuService {
+
+    @Autowired
```
가능한 Autowired 대신 constructor injection 을 사용해주세요

f-lab-joseph commented on src/main/java/com/yummy/delivery/service/MenuService.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    public void saveInitialTime(Menu menu){
+        menu.setCreated_at(LocalDateTime.now());
```
CAMEL CASE

log1995 committed:
CAMEL CASE 해결 및 의존성 주입방식 변경
  - mybatis-config.xml으로 CAMEL CASE 해결
  - @RequiredArgsConstructor 어노테이션을 이용하여 필드 주입 방식에서 생성자 주입방식으로 변경

log1995 committed:
일부 코드 수정
  - SellerController, SellerService 의존성 주입 방식 수정(생성자 주입)
  - 코드 내 미사용 import 구문 삭제

log1995 committed:
confilct 로 인한 코드 수정

log1995 committed:
conflict 로 인한 코드 수정

log1995 committed:
Merge remote-tracking branch 'origin/feature/15-menu-register' into feature/15-menu-register

log1995 committed:
Domain 패키지 내 클래스들의 변수를 primitive type에서 Wrapper type으로 변경

f-lab-joseph commented:
머지하겠습니다.
